src/gaxios.ts

```diff
@@ -254,6 +256,10 @@ export class Gaxios implements FetchCompliance {
     opts: GaxiosOptionsPrepared,
     res: Response,
   ): Promise<ReturnType<JSON['parse']>> {
+    if (res.status === HTTP_STATUS_NO_CONTENT) {
+      return '';
```

Comment:
why not an empty JSON here instead?

Reply:
This emulates the previous behavior before #617
If you return an empty object (`{}`), you are breaking backwards compatibility.

Reply:
Another good reason to return `null` or empty string is library user code checking if the `data` property is set.